src/components/link/link.style.ts

```diff
@@ -1,95 +1,210 @@
 import styled, { css } from "styled-components";
 import baseTheme from "../../style/themes/base";
 import StyledIcon from "../icon/icon.style";
-import { LinkProps } from "./link.component";
 
-interface StyledLinkProps
-  extends Pick<LinkProps, "isSkipLink" | "iconAlign" | "disabled"> {
+type Variants = "default" | "negative" | "neutral";
+export interface StyledLinkProps {
+  /** The disabled state of the link. */
+  disabled?: boolean;
+  /** Which side of the link to the render the link. */
+  iconAlign?: "left" | "right";
+  /** Allows to create skip link */
+  isSkipLink?: boolean;
+  /** Sets the colour styling when component is rendered on a dark background */
+  isDarkBackground?: boolean;
+  /** Allows link styling to be updated for light or dark backgrounds */
+  variant?: Variants;
+}
+interface PrivateStyledLinkProps {
   hasContent: boolean;
 }
 
-const StyledLink = styled.span<StyledLinkProps>`
-  ${({ isSkipLink, theme, iconAlign, hasContent, disabled }) => css`
-    display: inline-block;
-
-    ${isSkipLink &&
-    css`
-      a {
-        position: absolute;
-        padding-left: 24px;
-        padding-right: 24px;
-        line-height: 36px;
-        left: -999em;
-        z-index: ${theme.zIndex.aboveAll};
-        box-shadow: inset 0 0 0 2px var(--colorsActionMajor500);
-        border: 2px solid var(--colorsUtilityYang100);
-      }
+interface LinkColors {
+  color: string;
+  hoverColor: string;
+  disabledColor: string;
+}
 
-      a:focus {
-        top: 8px;
-        left: 8px;
-      }
-    `}
-
-    a,
-  button {
-      font-size: ${isSkipLink ? "16px" : "14px"};
-      text-decoration: underline;
-      color: ${isSkipLink
-        ? "var(--colorsUtilityYin090)"
-        : "var(--colorsActionMajor500)"};
+type ColorMapProp = (variant?: Variants) => LinkColors;
+
+interface ColorMap {
+  light: ColorMapProp;
+  dark: ColorMapProp;
+}
+
+const colorMap: ColorMap = {
+  light: (variant) => {
+    let color = "var(--colorsActionMajor500)";
+    let hoverColor = "var(--colorsActionMajor600)";
+
+    if (variant === "negative") {
+      color = "var(--colorsSemanticNegative500)";
+      hoverColor = "var(--colorsSemanticNegative600)";
+    } else if (variant === "neutral") {
+      color = "var(--colorsActionMajorYin090)";
+    }
+
+    return {
+      color,
+      hoverColor,
+      disabledColor: "var(--colorsActionMajorYin030)",
+    };
+  },
+  dark: (variant) => {
+    let color = "var(--colorsActionMajor350)";
+    let hoverColor = "var(--colorsActionMajor450)";
+
+    if (variant === "negative") {
+      color = "var(--colorsSemanticNegative350)";
+      hoverColor = "var(--colorsSemanticNegative450)";
+    } else if (variant === "neutral") {
+      color = "var(--colorsActionMinor100)";
+    }
+
+    return {
+      color,
+      hoverColor,
+      disabledColor: "var(--colorsActionMajorYang030)",
```

Comment:
The style changes to `Link` look good to me! Unfortunately we do have a colour contrast violation for the disabled link on a dark background @edleeks87. @harpalsingh could we use a different temporary value for now until the token is updated, or is there an alternative token we could use?

![Screenshot 2022-09-12 at 13 45 38](https://user-images.githubusercontent.com/18368713/189657152-96664254-56a1-4763-acb8-6e9b4c2fe6b6.png)

Reply:
Good spot, sorry I meant to add something to the PR description. DS are aware of the issue and will be reviewing the approach at some point but for now have asked to implement as is defined in the figma file